cellprofiler_core/modules/metadata.py

```diff
@@ -1161,15 +1151,43 @@ def msg(url):
                     if metadata is None:
                         metadata = get_omexml_metadata(url=url)
                         filelist.add_metadata(url, metadata)
-                except:
-                    print("Metadata extraction failed for %s" % url)
+                except Exception as e:
+                    import logging
+                    logger = logging.getLogger(__name__)
+                    logger.exception(f"Metadata extraction failed for {url}", e)
                     errorcount += 1
-            if errorcount > 0:
-                wx.MessageBox(
-                    "Metadata extraction completed with %i error(s).\n"
-                    "File(s) may be in an incompatible format or "
-                    "corrupted." % errorcount
-                )
+                if errorcount > 0:
+                    msg = f"Metadata extraction completed with {errorcount} error(s).\n"
+                    "File(s) may be in an incompatible format or corrupted."
+                    if dlg is not None:
+                        import wx
+                        wx.MessageBox(msg)
+                    else:
+                        raise ValueError(msg)
+
+        is_headless = get_headless()
```

Comment:
All of this is so that regular UI (not headless) isn't affected and we can merge upstream.